File: apps/web/src/types/source.ts
export type SourceStatus =
  | 'active'
  | 'requires_key'
  | 'manual_import'
  | 'planned'
  | 'research'
  | 'not_implemented'
  | 'test_fixture_only'

export type SourceCategory =
  | 'electricity'
  | 'weather'
  | 'ev'
  | 'environmental'
  | 'resilience'
  | 'gis'
  | 'regulatory'

export interface Source {
  id: string
  name: string
  description: string
  category: SourceCategory
  status: SourceStatus
  url?: string
  update_frequency?: string
  data_format?: string
  notes?: string
  requires_api_key?: boolean
  api_key_env_var?: string
  auth_type?: string
  access_type?: string
  data_geography?: string
  connector_implemented?: boolean
  phase_added?: number
  last_verified?: string
  is_real_data?: boolean
  is_mock_data?: boolean
}

export interface SourceList {
  sources: Source[]
  total: number
}

File: apps/api/app/schemas/source.py
from enum import Enum
from typing import List, Optional

from pydantic import BaseModel


class SourceStatus(str, Enum):
    active = "active"
    unavailable = "unavailable"
    mock = "mock"
    planned = "planned"


class SourceCategory(str, Enum):
    electricity = "electricity"
    weather = "weather"
    ev = "ev"
    environmental = "environmental"
    resilience = "resilience"
    gis = "gis"
    regulatory = "regulatory"


class Source(BaseModel):
    id: str
    name: str
    description: str
    category: SourceCategory
    status: SourceStatus
    url: Optional[str] = None
    update_frequency: Optional[str] = None
    data_format: Optional[str] = None
    notes: Optional[str] = None


class SourceList(BaseModel):
    sources: List[Source]
    total: int
